prow/jenkins/metrics.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jenkins
+
+import "github.com/prometheus/client_golang/prometheus"
+
+var (
+	requests = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "jenkins_requests",
+		Help: "Number of Jenkins requests made from prow.",
+	}, []string{
+		// http verb of the request
+		"verb",
+		// path of the request
+		"handler",
+		// http status code of the request
+		"code",
+	})
+	requestRetries = prometheus.NewCounter(prometheus.CounterOpts{
```

Comment:
The retries could also be labeled with `verb` and `handler`.

Reply:
I want to avoid more metrics for now (every combination of labels is its own metric). We are going to reiterate on these anyway.